src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfig.java

```diff
@@ -55,65 +53,102 @@
+    public MultiLangDaemonConfig(String propertiesFile, ClassLoader classLoader,
             KinesisClientLibConfigurator configurator) throws IOException, IllegalArgumentException {
         Properties properties = loadProperties(classLoader, propertiesFile);
         if (!validateProperties(properties)) {
-            throw new IllegalArgumentException("Must provide an executable name in the properties file, "
-                    + "e.g. executableName = sampleapp.py");
+            throw new IllegalArgumentException(
+                    "Must provide an executable name in the properties file, " + "e.g. executableName = sampleapp.py");
         }
 
         String executableName = properties.getProperty(PROP_EXECUTABLE_NAME);
         String processingLanguage = properties.getProperty(PROP_PROCESSING_LANGUAGE);
+        ClientConfiguration clientConfig = buildClientConfig(properties);
+
+        kinesisClientLibConfig = configurator.getConfiguration(properties).withKinesisClientConfig(clientConfig)
+                .withCloudWatchClientConfig(clientConfig).withDynamoDBClientConfig(clientConfig);
 
-        kinesisClientLibConfig = configurator.getConfiguration(properties);
         executorService = buildExecutorService(properties);
-        recordProcessorFactory = new MultiLangRecordProcessorFactory(executableName, executorService, kinesisClientLibConfig);
+        recordProcessorFactory = new MultiLangRecordProcessorFactory(executableName, executorService,
+                kinesisClientLibConfig);
 
         LOG.info("Running " + kinesisClientLibConfig.getApplicationName() + " to process stream "
                 + kinesisClientLibConfig.getStreamName() + " with executable " + executableName);
         prepare(processingLanguage);
     }
 
+    private ClientConfiguration buildClientConfig(Properties properties) {
+        ClientConfiguration clientConfig = new ClientConfiguration();
+        String proxyHost = "";
+        int proxyPort = 0;
+
+        if (properties.getProperty("http.proxyHost") != null) {
+            LOG.info("Getting proxy info from properties file.");
+
+            proxyHost = properties.getProperty("http.proxyHost");
+            proxyPort = Integer.parseInt(properties.getProperty("http.proxyPort"));
+        } else if (System.getProperty("http.proxyHost") != null) {
+            LOG.info("Getting proxy info from java system properties");
+
+            proxyHost = System.getProperty("http.proxyHost");
+            proxyPort = Integer.parseInt(System.getProperty("http.proxyPort"));
+        } else if (System.getenv("http_proxy") != null) {
+            LOG.info("Getting proxy info environment settings");
+
+            try {
+                URI proxyAddr = new URI(System.getenv("http_proxy"));
+
+                proxyHost = proxyAddr.getHost();
+                proxyPort = proxyAddr.getPort();
+            } catch (URISyntaxException e) {
+                LOG.error("System proxy not set correctly");
```

Comment:
Could you log the error too

Reply:
will do

Reply:
You can set the exception in the same log message,

```java
LOG.error("Message", e)
```